fireatlas/postprocess.py

```diff
@@ -261,38 +263,80 @@ def save_fire_layers(allfires_gdf_fid, region, fid, tst):
 
         data.to_file(os.path.join(output_dir, f"{layer}.fgb"), driver="FlatGeobuf")
 
+@timed
+def fill_activefire_rows(allfires_gdf, ted):
+    """For a subset of allfires data, add any rows where a fire is still
+    active, but is not burning.
+    """
+    dd = singlefire_getdd("all")
+    
+    if allfires_gdf.index.names == ['fireID', 't']:
+        gdf = allfires_gdf.reset_index()
+    else:
+        gdf = allfires_gdf
+
+    dt = t2dt(ted)
+    all_new_rows = []
+    for fid, allfires_gdf_fid in gdf.groupby("fireID"):
+        d = allfires_gdf_fid.set_index("t")
+        
+        # set last timestep so we fill after the fire has stopped burning
+        if not d.iloc[-1]["invalid"]:
+            last_t = d.index[-1]
+            if last_t != dt:
+                last_t = min(last_t + datetime.timedelta(days=settings.maxoffdays), dt)
+                d.loc[last_t] = None
+
+        ffilled = d.resample("12H").ffill(limit=settings.limoffdays*2).dropna(how="all")
+
+        # get all the rows that are new
+        new_rows = ffilled[~ffilled.index.isin(d.index)]
+
+        # set values that should not be forward filled.
+        new_rows.loc[:,["n_newpixels", "meanFRP", "nfp"]] = 0, None, None
+        
+        all_new_rows.append(new_rows.reset_index())
+            
+    output = pd.concat([gdf, *all_new_rows]).sort_values(["t", "fireID"])
+    for k, tp in dd.items():
+        output[k] = output[k].astype(tp)
+    return output
 
 @timed
 def merge_rows(allfires_gdf_fid):
     """For a subset of allfires data containing only one fire, merge any
     rows that have the same `t`
     """
-    output = allfires_gdf_fid.drop_duplicates(subset=["t"]).set_index("t").copy()
-
-    # clean up any merges that are needed
-    for dt, rows in allfires_gdf_fid[allfires_gdf_fid.t.duplicated(False)].groupby("t"):
-        # first get the weighted sums for pixden and meanFRP
-        pixweight = (rows["pixden"] * rows["farea"]).sum()
-        FRPweight = (rows["meanFRP"] * rows["n_pixels"]).sum()
-
-        for col in ["n_pixels", "n_newpixels", "farea", "fperim", "flinelen"]:
-            output.loc[dt, col] = rows[col].sum()
-
-        output.loc[dt, "t_st"] = rows["t_st"].min()
-        output.loc[dt, "pixden"] = pixweight / output.loc[dt, "farea"]
-        output.loc[dt, "meanFRP"] = FRPweight / output.loc[dt, "n_pixels"]
-
-        dissolved = rows.dissolve()
-        for col in ["hull", "fline", "nfp"]:
-            output.loc[dt, col] = dissolved[col].item()
-
+    output = allfires_gdf_fid.dissolve(
+        by="t",
+        aggfunc={
+            "meanFRP": lambda x: (
+                None 
+                if (n_newpixels := allfires_gdf_fid.loc[x.index, "n_newpixels"]).sum() == 0 
+                else (x * n_newpixels).sum() / n_newpixels.sum()
+            ),
+            "n_newpixels": "sum",
+            "duration": "max",
```

Comment:
`max` is not enough since any individual fire being combined won't necessarily last the whole time. So better to calculate duration directly something like: `(gdf.t.min() - gdf.t.iloc[0]).dt.days`